src/puppetlabs/puppetdb/queue.clj

```diff
@@ -112,24 +125,24 @@
   certname if the certname is long or contains filesystem special characters."
   ([] (metadata-serializer puppetdb-command->metadata-command))
   ([puppetdb-command->metadata-command]
-   (fn [received command version certname]
+   (fn [received {:keys [producer-ts command version certname]}]
      (when-not (puppetdb-command->metadata-command command)
        (throw (IllegalArgumentException. (trs "unknown command ''{0}''" command))))
      (let [certname (or certname "unknown-host")
-           recvd-long (tcoerce/to-long received)
+           received+producer-time (received+producer-ts received producer-ts)
            short-command (puppetdb-command->metadata-command command)
            safe-certname (embeddable-certname received short-command version certname)]
        (if (= safe-certname certname)
-         (format "%d_%s_%d_%s.json" recvd-long short-command version safe-certname)
+         (format "%s_%s_%d_%s.json" received+producer-time short-command version safe-certname)
          (let [name-hash (kitchensink/utf8-string->sha1 certname)]
-           (format "%d_%s_%d_%s_%s.json"
-                   recvd-long short-command version safe-certname name-hash)))))))
+           (format "%s_%s_%d_%s_%s.json"
+                   received+producer-time short-command version safe-certname name-hash)))))))
 
 (def serialize-metadata (metadata-serializer))
 
 (defn metadata-rx [valid-commands]
   (re-pattern (str
-               "([0-9]+)_"
+               "([0-9]+)([+|-][0-9]+)?_"
```

Comment:
Also yes, it will be omitted from the filename. We won't have producer timestamps for old messages in the queue (or AMQ), imports etc.